lib/ensure/file.js

```diff
@@ -8,9 +8,21 @@ const pathExists = require('../path-exists').pathExists
 
 function createFile (file, callback) {
   function makeFile () {
-    fs.writeFile(file, '', err => {
+    const dir = path.dirname(file)
+
+    fs.stat(dir, (err, stats) => {
       if (err) return callback(err)
-      callback()
+      if (!stats.isDirectory()) {
+        // This is just to cause an internal ENOTDIR error to be thrown
+        fs.readdir(dir, err => {
+          if (err) return callback(err)
+        })
+      } else {
+        fs.writeFile(file, '', err => {
+          if (err) return callback(err)
+          callback()
+        })
+      }
```

Comment:
Not too happy with this, seems a bit unreadable. It's been a while since I've worked with callbacks. Cleanup suggestions welcome.